EpiAware/src/EpiObsModels/StackObservationModels.jl

```diff
@@ -0,0 +1,112 @@
+@doc raw"
+
+A stack of observation models that are looped over to generate observations for
+each model in the stack. Note that the model names are used to prefix the parameters in each model (so if I have a model named `cases` and a parameter `y_t`, the parameter in the model will be `cases.y_t`).
+
+## Constructors
+
+- `StackObservationModels(models::Vector{<:AbstractTuringObservationModel},
+    model_names::Vector{<:AbstractString})`: Construct a `StackObservationModels` object with a vector of observation models and a vector of model names.
+    - `StackObservationModels(; models::Vector{<:AbstractTuringObservationModel},
+    model_names::Vector{<:AbstractString})`: Construct a `StackObservationModels` object with a vector of observation models and a
+    vector of model names.
+- `StackObservationModels(models::Vector{<:AbstractTuringObservationModel})`:
+    Construct a `StackObservationModels` object with a vector of observation models. The model names are automatically generated from the observation model types.
```

Comment:
Do we really want to support this constructor? To get it to work you need data like `(PoissonError = missing, NegativeBinomialError = missing)`

Reply:
Its not totally clear to me. I would be happy to take it out as we are unlikely to want to use it.

Reply:
I would support leaving it out because just passing a vector of models _without_ names and then getting confused would be something I'd expect to do! I'd rather have an error about not having supplied names at the moment of construction

Reply:
Saw this done